src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -16,25 +12,34 @@
 using System.Threading;
 using System.Threading.Tasks;
 using System.Xml;
-#if MANAGED_SNI
-using System.Data.SqlClient.SNI;
-#endif
+
 
 using MSS = Microsoft.SqlServer.Server;
 
 namespace System.Data.SqlClient
 {
+
+    struct SNIErrorDetails
+    {
+        public string errorMessage;
+        public uint nativeError;
+        public uint sniErrorNumber;
+        public int provider;
+        public uint lineNumber;
+        public string function;
+        public Exception exception;
+    }
+
     // The TdsParser Object controls reading/writing to the netlib, parsing the tds,
     // and surfacing objects to the user.
-    sealed internal class TdsParser
+    sealed internal partial class TdsParser
     {
         static TdsParser()
         {
             // For CoreCLR, we need to register the ANSI Code Page encoding provider before attempting to get an Encoding from a CodePage
             Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
```

Comment:
I think registering the encoding provider should be the app decision and not the library decision. having the library registering the provider means any app using the library will pay the cost of the encoding even if the app not using the non-Unicode encodings. 

@stephentoub encoding handling didn't change in Net core 2.0. the only change we did is we have made the console support non-Unicode encodings but this is limited to the console only.

---
In reply to: [103917083](https://github.com/dotnet/corefx/pull/16589#discussion_r103917083) [](ancestors = 103917083)